### ISSUE TITLE
Vagrant failing when using ssh agent

### ISSUE THREAD

**Issue by laszlof** [mitchellh/vagrant#7005]
We've recently noticed an issue when enabling agent forwarding in vagrant. It seems that SSH commands executed inside the guests will randomly fail when this option is enabled. It seems to happen anywhere from mounting the synced folders (/vagrant, for example), to executing shell provisioners. It seems completely random when it happens, But I have been able to reproduce it in a small scale taking out any additional things that may cause issues.

Here are the versions I am using:

Vagrant 1.8.1
VirtualBox 5.0.14
Ubuntu 15.10

I've seen several other issues that seem to stem around this same problem. I've include a gist below that should allow someone to reproduce the issue. The only setup requirements are having an ssh key installed locally that is setup in github.

Let me know if you need any further information.

https://gist.github.com/laszlof/1ee34c32d19ce404c5a9

**Comment by laszlof** [comment]
Small update on this. I tried using the following boxes with this setup:

- bento/centos-7.1 (50% FAILURE)
- centos/7 (60% FAILURE)
- hashicorp/precise64 (100% SUCCESS!)

I'm trying a few more now, but it looks like this might be specific to centos 7 boxes. Will report back as I get more information.

**Comment by kostyukdg** [comment]
I have this problem too

Vagrant 1.8.1
VirtualBox 5.0.14
Mac OS X 10.11.1

**Comment by laszlof** [comment]
A bit more information on this. It seems that its specific to CentOS 7.1 boxes. However, I did find one that worked (a box built from packer). Theres likely something in the default configuration of CentOS 7.1 thats cause breakage here.

@kostyukdg Have you tried using another base box?

**Comment by sethvargo** [comment]
Hi @laszlof 

Thank you for opening an issue. I don't think this is a Vagrant bug, but more likely an OS+virtualbox bug. It's likely that the guest addition versions are incorrect and that's causing weird network outages. Once Vagrant connects to the machine via SSH, its work is really done. Everything beyond that is the OS and the virtualization layer itself. The fact that it's more prevalent on "newer" operating system really confirms that suspicion.

**Comment by laszlof** [comment]
I actually tracked this down to a specific default setting in the base image, I don't recall what it was. I'll have to look on my work machine.

**Comment by sethvargo** [comment]
Okay - if you could share, I can let the bento team know

**Comment by laszlof** [comment]
Ok. So it looks like I lied. We ended up building our own image based off of a packer image we found, and it worked like a charm. I'm not really sure what the difference was, as it appeared to be fine and the configuration was basically the same as bento builds.

**Comment by laszlof** [comment]
For reference, here are the packer files we're using:

https://github.com/laszlof/stark-packer

**Comment by sethvargo** [comment]
Hi there,

I just tried those boxes from Atlas multiple times, with ssh-agent enabled, and I've been unable to reproduce this once. I'm on a Mac host, so it's possible this is something odd with SSH on Ubuntu, but we specifically tell Vagrant to ignore the ssh-agent (in fact, I just merged a patch to make this customizable), so unfortunately I'm out of ideas here. 

I'll leave this open for a bit, but if this isn't a widespread issue, I'm not sure how to fix it. Sorry!
